yt_dlp/extractor/japandiet.py

```diff
@@ -0,0 +1,259 @@
+import re
+
+from ..utils import (
+    ExtractorError,
+    clean_html,
+    float_or_none,
+    int_or_none,
+    join_nonempty,
+    smuggle_url,
+    traverse_obj,
+    unsmuggle_url
+)
+from .common import InfoExtractor
+
+
+class ShugiinItvBaseIE(InfoExtractor):
+    @classmethod
+    def _find_rooms(cls, webpage):
+        return [{
+            '_type': 'url',
+            'id': x.group(1),
+            'title': clean_html(x.group(2)).strip(),
+            'url': smuggle_url(f'https://www.shugiintv.go.jp/jp/index.php?room_id={x.group(1)}', x.groups()),
+            'ie_key': ShugiinItvLiveIE.ie_key(),
+        } for x in re.finditer(r'<a\s+href=".+?\?room_id=(room\d+)"\s*class="play_live".+?class="s12_14">(.+?)</td>', webpage)]
```

Comment:
```suggestion
        } for x in re.finditer(r'<a\s+href="[^"]+\?room_id=(room\d+)"\s*class="play_live".+?class="s12_14">(.+?)</td>', webpage)]
```

The regex looks fragile. Though I haven't looked at the page, so not sure if there's a better way